ethereumetl/service/token_transfer_v2_extractor.py

```diff
@@ -80,8 +80,17 @@ def extract_transfer_from_log(self, receipt_log):
             token_ids = topics_with_data[7: 7 + size]
             amounts = topics_with_data[1 + 7 + size:]
 
-            token_transfers = []
+            # group by token_id and sum amounts to handle duplicate token_ids in same batch transfer
+            # this breaks primary key constraint in database
+            # https://rinkeby.etherscan.io/tx/0x08ea1641fde8e29fbeea762eb28fd9bc4bcfc1ae6e31844c50005b1a8cf91693#eventlog
+            token_id_to_amounts_dec = {}
             for token_id, amount in zip(token_ids, amounts):
+                amount_dec = hex_to_dec(amount)
+                token_id_to_amounts_dec[token_id] = (token_id_to_amounts_dec.get(token_id, 0) + amount_dec)
+
+            token_transfers = []
+            for token_id, amount_dec in token_id_to_amounts_dec.items():
+                amount = "0x{:064x}".format(amount_dec)
```

Comment:
ideally I would like to emit human friendly decimals in kafka events/ tables. But dont want to break current flows. So working around by padding for 64 bit numbers.

Reply:
usually storing as a hex string is easier since the u256 size is larger than most database integer types. Postgres only supports  up to `9223372036854775807` (`bigint`) for example. we'll definitely have token ids bigger than this